fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/derby/DerbyMaster.java

```diff
@@ -166,75 +213,122 @@ public IDatabaseTranslator getTranslator() {
      * Ask the schema to apply itself to our target (adapter pattern)
      * @param pdm
      */
-    public void createSchema(PhysicalDataModel pdm) {
-        createSchema(vhs, pdm);
+    public void createSchema(IConnectionProvider pool, PhysicalDataModel pdm) {
+        createSchema(pool, vhs, pdm);
     }
 
     /**
      * Ask the schema to apply itself to our target (adapter pattern)
      * @param vhs
      * @param pdm
      */
-    public void createSchema(IVersionHistoryService vhs, PhysicalDataModel pdm) {
-        runWithAdapter(target -> pdm.applyWithHistory(target, vhs));
+    public void createSchema(IConnectionProvider pool, IVersionHistoryService vhs, PhysicalDataModel pdm) {
+        runWithAdapter(pool, target -> pdm.applyWithHistory(target, vhs));
     }
+    
+    public void runWithAdapter(IConnectionProvider pool, java.util.function.Consumer<IDatabaseAdapter> fn) {
```

Comment:
Why the full package name is specified?

Reply:
Probably to help Eclipse find the right Consumer more quickly when typing. No other reason.

Reply:
added import